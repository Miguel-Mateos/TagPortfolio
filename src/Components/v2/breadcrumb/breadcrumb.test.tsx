import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Breadcrumb, {IBreadcrumbProps, BreadcrumbItem} from './breadcrumb';

const BreadcrumbExample = (props?: IBreadcrumbProps) => {
    return (
        <Breadcrumb {...props} className="tag-ds breadcrumb-wrapper" data-testid="breadcrumb-test">
            <BreadcrumbItem href="Breadcrumb_1" data-testid="breadcrumb-1" id="breadcrumb-1" title="Breadcrumb Test 1"></BreadcrumbItem>
            <BreadcrumbItem href="Breadcrumb_2" data-testid="breadcrumb-2" id="breadcrumb-2" title="Breadcrumb Test 2"></BreadcrumbItem>
            <BreadcrumbItem href="Breadcrumb_3" data-testid="breadcrumb-3" id="breadcrumb-3" title="Breadcrumb Test 3"></BreadcrumbItem>
            <BreadcrumbItem href="Breadcrumb_4" data-testid="breadcrumb-4" id="breadcrumb-4" title="Breadcrumb Test 4"></BreadcrumbItem>
            <BreadcrumbItem href="Breadcrumb_5" data-testid="breadcrumb-truncated" id="breadcrumb-5" title="Breadcrumb title with more than 30 characters"></BreadcrumbItem>
        </Breadcrumb>
)}

test('render Breadcrumb and match snap', () => {
    const component = renderer.create(<BreadcrumbExample />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('display a dropdown menu when there is more than 4 items', () => {
    const {container} = render(<BreadcrumbExample />);

    expect(container.getElementsByClassName('breadcrumb-menu-container'));
});

test('display breadcrumb text truncated when it has more than 30 characters', () => {
    const {getByTestId} = render(<BreadcrumbExample />);
    const breadcrumbToTruncate = getByTestId('breadcrumb-truncated-link');

    expect(breadcrumbToTruncate).toHaveTextContent('Breadcrumb title with more tha...');
});


