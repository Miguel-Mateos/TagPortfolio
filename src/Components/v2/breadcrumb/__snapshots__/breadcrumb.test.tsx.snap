// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Breadcrumb and match snap 1`] = `
<div
  className="breadcrumb-container tag-ds breadcrumb-wrapper"
  data-testid="breadcrumb-test"
>
  <div
    className="breadcrumb-item false"
    data-testid="breadcrumb-1"
    id="Breadcrumb Test 1"
    onClick={[Function]}
  >
    <a
      className="breadcrumb-item"
      data-testid="breadcrumb-1-link"
      href="Breadcrumb_1"
      id="breadcrumb-1"
      target="_self"
    >
      Breadcrumb Test 1
    </a>
  </div>
  <span
    className="material-icons breadcrumb-separator"
  >
    chevron_right
  </span>
  <div
    className="dropdown breadcrumb-menu-container"
  >
    <div
      className="dropdown-button "
      onClick={[Function]}
    >
      <span
        className=""
      />
      <span
        className="material-icons breadcrumb-dots"
      >
        more_horiz
      </span>
    </div>
    <div
      className="dropdown-menu  breadcrumb-menu-content "
    >
      <li>
        <div
          className="tag-ds dropdown-item tab-list-item false"
          data-testid="breadcrumb-2"
          id="Breadcrumb Test 2"
          onClick={[Function]}
        >
          <a
            className="breadcrumb-item"
            data-testid="breadcrumb-2-link"
            href="Breadcrumb_2"
            id="breadcrumb-2"
            target="_self"
          >
            Breadcrumb Test 2
          </a>
        </div>
        <div
          className="tag-ds dropdown-item tab-list-item false"
          data-testid="breadcrumb-3"
          id="Breadcrumb Test 3"
          onClick={[Function]}
        >
          <a
            className="breadcrumb-item"
            data-testid="breadcrumb-3-link"
            href="Breadcrumb_3"
            id="breadcrumb-3"
            target="_self"
          >
            Breadcrumb Test 3
          </a>
        </div>
        <div
          className="tag-ds dropdown-item tab-list-item false"
          data-testid="breadcrumb-4"
          id="Breadcrumb Test 4"
          onClick={[Function]}
        >
          <a
            className="breadcrumb-item"
            data-testid="breadcrumb-4-link"
            href="Breadcrumb_4"
            id="breadcrumb-4"
            target="_self"
          >
            Breadcrumb Test 4
          </a>
        </div>
      </li>
    </div>
  </div>
  <span
    className="material-icons breadcrumb-separator"
  >
    chevron_right
  </span>
  <div
    className="breadcrumb-item breadcrumb-current"
    data-testid="breadcrumb-truncated"
    id="Breadcrumb title with more than 30 characters"
    onClick={[Function]}
  >
    <a
      className="breadcrumb-item"
      data-testid="breadcrumb-truncated-link"
      href="Breadcrumb_5"
      id="breadcrumb-5"
      target="_self"
    >
      Breadcrumb title with more tha...
    </a>
  </div>
</div>
`;
